test/index.js

```diff
@@ -445,7 +805,27 @@ describe('def', function() {
   });
 
   it('supports custom types', function() {
-    var env = $.env.concat([Integer, $Pair]);
+
+    //  anonJust :: a -> Just a
+    var anonJust = function(x) {
+      return {
+        '@@type': 'my-package/anonMaybe',
+        'constructor': anonJust,  // ramda/ramda#1516
```

Comment:
Let's name the function `anonJust` (in addition to assigning it to a variable of that name). It doesn't seem right to reference a variable in outer scope (which does not even have the intended value at the time of this function's creation). Self-referential functions should be named functions, I believe.